### PR TITLE
Create npm-publish.yml

### PR DESCRIPTION
**Pull request by kaiwerther**
Added automatic publish to npm on commit.

By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

replaces manual step for deployment

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# On each new tag deploy to npm
```
Comment (tishoyanchev) [inline .github/workflows/npm-publish.yml:1]: This text is not clear enough, and can be easily misinterpreted.
Please replace it with the following:
"Every time a tag is pushed, it gets automatically published to npm".

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# this also means that new tags should always have a new version number in package.json
+# they are best created with npm version
```
Comment (tishoyanchev) [inline .github/workflows/npm-publish.yml:3]: Please clarify what you mean by this - "they are best created with npm version"

**Review by tishoyanchev — COMMENTED** [review]
In addition to the two comments that I left, do you think it would be better if the publish is triggered only on specific repositories, such as master / develop?

**Comment by julrich** [comment]
As a suggestion: I'd use something like `auto` (https://github.com/intuit/auto) for automatic versioning. Makes it configurable which branches get what kind of versions, and updates according to PR labels. Can still be used / triggered through Github actions!

Stable new versions should be generated from the `main` / `master` branch only, imo. Other branches (like `next` / `develop`, etc) should use corresponding pre-release configurations (like releasing release candidates or canary releases).

Proved to be super efficient and flexible for us (even supports creating support branches for older versions that need bugfix releases, something not to be overlooked).

**Comment by kaiwerther** [comment]
Looks promissing, will look into it - I will close this PR

**Pull request closed without merging**